### PR TITLE
Update Rtos.md - DO NOT MERGE (before mbed-os 5.7 release)

### PR DESCRIPTION
**Pull request by deepikabhavnani**
Removed all RTX mention and added reference to importer script

### CODE REVIEW

**Comment by deepikabhavnani** [comment]
@sg- RTX references are removed. As per jira: There is lots of good content for api/rtos/rtos (not relevant to porting). Please list what else should be covered in rtos.md page.

**Comment by AnotherButler** [comment]
@deepikabhavnani We recently updated this, so the 5.6 branch is the live 5.6 docs on the web, and this new_engine branch is the working branch that we will create 5.7 from. Is it OK if I remove the "code dependency" label and merge this after @sg- reviews it?

**Comment by deepikabhavnani** [comment]
Yes @AnotherButler we can merge it after review.

File: docs/reference/contributing/target/rtos.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### RTOS
 
-CMSIS/RTX code is imported from <a href="https://github.com/ARM-software/CMSIS_5/" target="_blank">the original CMSIS repository</a>.
+CMSIS/RTX code is imported from <a href="https://github.com/ARM-software/CMSIS_5/" target="_blank">the original CMSIS repository</a> with the help of Python script in the `mbed-os` repository `mbed-os\tools\importer\importer.py`.
```
Comment (sg-) [inline docs/reference/contributing/target/rtos.md]: original should be 'upstream development'

Comment (sg-) [reply docs/reference/contributing/target/rtos.md]: further down there is generic system would should be 'platform operating system'

File: docs/reference/contributing/target/rtos.md [hunk 2]
```diff
@@ -25,51 +25,8 @@ Option | Value | Description |
...
-`CMSIS_5/CMSIS/Core/Include/cmsis_compiler.h` | `mbed-os/cmsis/` | Toolchain generic code |
-`CMSIS_5/CMSIS/Core/Include/cmsis_{armcc,armclang,gcc}.h` | `mbed-os/cmsis/TOOLCHAIN_{ARM,GCC}/` | Toolchain specific code |
-`CMSIS_5/CMSIS/RTOS2/Include/cmsis_os2.h` | `mbed-os/rtos/TARGET_CORTEX/rtx5/` | RTX main header |
-`CMSIS_5/CMSIS/RTOS2/RTX/Config/` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX configuration files |
-`CMSIS_5/CMSIS/RTOS2/RTX/Include1/` | `mbed-os/rtos/TARGET_CORTEX/rtx4` | RTOS1 compatibility layer |
-`CMSIS_5/CMSIS/RTOS2/RTX/Include/` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX definitions |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/rtx_*` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX sources |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/svc_user.c` | `mbed-os/rtos/rtx2/TARGET_CORTEX_M/` | RTX SVC user table |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/{ARM,GCC,IAR}/` | `mbed-os/rtos/TARGET_CORTEX/rtx5/TARGET_{M0,M0P,M3,RTOS_M4_M7,M23,M33}/TOOLCHAIN_{ARM,GCC,IAR}` | Toolchain and core specific exception handlers |
+Due to differences in how the Mbed OS and CMSIS directory structures look, you can't import the original code directly using `importer.py` and configuration file `cmsis_importer.json`.
```
Comment (sg-) [inline docs/reference/contributing/target/rtos.md]: This sentence seems wrong, you cant import using a tool we created?

**Comment by bulislaw** [comment]
@deepikabhavnani I think that got forgotten.

**Comment by deepikabhavnani** [comment]
@bulislaw - Changes are done, waiting for @sg- to review and approve

File: docs/reference/contributing/target/rtos.md [hunk 3]
```diff
@@ -1,75 +1,31 @@
...
+CMSIS/RTX code is imported from [the original CMSIS repository](https://github.com/ARM-software/CMSIS_5/) with the help of a Python script in the `mbed-os` repository `mbed-os\tools\importer\importer.py`.
 
 #### Memory considerations
 
-Please note that Arm Mbed OS doesn't use any of the RTX memory models, which are based on static carveouts (memory pools). This approach is not ideal for generic systems, such as Mbed OS, because calculating required numbers of RTOS objects is impossible. To avoid declaring arbitrary large buffers carved out at compile time, limiting the amount of available memory, Mbed OS shifts the responsibility of supplying the backing memory to CMSIS-RTOS2 users.
+Please note that Arm Mbed OS doesn't use any of the RTX memory models, which are based on static carveouts (memory pools). This approach is not ideal for platform operating systems, such as Mbed OS, because calculating required numbers of RTOS objects is impossible. To avoid declaring arbitrary large buffers carved out at compile time, limiting the amount of available memory, Mbed OS shifts the responsibility of supplying the backing memory to CMSIS-RTOS2 users.
```
Comment (bulislaw) [inline docs/reference/contributing/target/rtos.md]: I'm not sure what the `platform operating systems` are to be fair.

File: docs/reference/contributing/target/rtos.md [hunk 4]
```diff
@@ -1,75 +1,31 @@
...
 Option | Value | Description |
 -------|-------|-------------|
-`OS_STACK_SIZE` | 4K or 2K | For a normal target, the thread stack size is set to 4K; for constrained targets, it's 2K. |
+`OS_STACK_SIZE` | 4K | OS Stack size is set as `MBED_CONF_APP_THREAD_STACK_SIZE` which is 4096 as default. |
 `OS_TIMER_THREAD_STACK_SIZE` | 768B | Timer thread stack set to 768B that's necessary to support the C++ wrappers (4 instances), but it may require changing to support larger number of active timers. |
 `OS_IDLE_THREAD_STACK_SIZE` | 512B | Required to handle Mbed OS wrappers |
 `OS_DYNAMIC_MEM_SIZE` | 0 | RTX dynamic memory is disabled. |
-`OS_MUTEX_OBJ_MEM` | 1 or 0 | For ARMC, use 1; for other toolchains, it's 0. ARMC uses statically allocated mutexes internally. |
-`OS_MUTEX_NUM` | 6 or 0 | For ARMC, use 6; for other toolchains, it's 0. ARMC uses statically allocated mutexes internally. |
+`OS_TICK_FREQ` | 1000 | OS Tickrate must be 1000 for system timing. |
```
Comment (bulislaw) [inline docs/reference/contributing/target/rtos.md]: Can we change that note to something like: Mbed OS requires 1ms system tick?

File: docs/reference/contributing/target/rtos.md [hunk 5]
```diff
@@ -1,75 +1,31 @@
...
-`CMSIS_5/CMSIS/Core/Include/cmsis_{armcc,armclang,gcc}.h` | `mbed-os/cmsis/TOOLCHAIN_{ARM,GCC}/` | Toolchain specific code |
-`CMSIS_5/CMSIS/RTOS2/Include/cmsis_os2.h` | `mbed-os/rtos/TARGET_CORTEX/rtx5/` | RTX main header |
-`CMSIS_5/CMSIS/RTOS2/RTX/Config/` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX configuration files |
-`CMSIS_5/CMSIS/RTOS2/RTX/Include1/` | `mbed-os/rtos/TARGET_CORTEX/rtx4` | RTOS1 compatibility layer |
-`CMSIS_5/CMSIS/RTOS2/RTX/Include/` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX definitions |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/rtx_*` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX sources |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/svc_user.c` | `mbed-os/rtos/rtx2/TARGET_CORTEX_M/` | RTX SVC user table |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/{ARM,GCC,IAR}/` | `mbed-os/rtos/TARGET_CORTEX/rtx5/TARGET_{M0,M0P,M3,RTOS_M4_M7,M23,M33}/TOOLCHAIN_{ARM,GCC,IAR}` | Toolchain and core specific exception handlers |
+Due to differences in the Mbed OS and CMSIS directory structure, you can't import the original code directly. You should use the `importer.py` and configuration file `cmsis_importer.json` to import upstream CMSIS code.
```
Comment (bulislaw) [inline docs/reference/contributing/target/rtos.md]: I think right now it's mostly down to build system.

File: docs/reference/contributing/target/rtos.md [hunk 6]
```diff
@@ -1,75 +1,31 @@
...
-`rtx_lib.h` | Doxygen added; added per-thread uVisor context |
-`rtx_os.h` | Doxygen added; added per-thread uVisor context |
-`irq_cm4.s` | For all toolchains: added case for Cortex M4 cores without VFP |
-`svc_user.c` | Removed as its template file and should not be in our code base |
-`rt_OsEventObserver.{c,h}` | Added an interface for uVisor to be notified about certain events from privileged code |
-`irq_armv8mbl.S` | IAR toolchain: added file for Cortex M23 core |
-`irq_armv8mml.S` | IAR toolchain: added file for Cortex M33 core |
-
-<span class="notes">**Note:** For all toolchains, Mbed OS uses `irq_cm0.s` for both M0 and M0P cores.</span>
+Due to different use cases between Mbed OS and CMSIS, we modified the source code. We upstream our changes to the CMSIS repository, but in cases when they aren't compatible with CMSIS requirements, we maintain a small set of changes. We maintain changes as separate commits in `mbed-os`, and SHAs are in the `commit_sha` section of the `cmsis_importer.json` file.
```
Comment (bulislaw) [inline docs/reference/contributing/target/rtos.md]: > Due to different use cases between Mbed OS and CMSIS, we modified the source code

that's a bit vogue, which source code?

> We upstream our changes to the CMSIS repository, but in cases when they aren't compatible with CMSIS requirements, we maintain a small set of changes. We maintain changes as separate commits in `mbed-os`, and SHAs are in the `commit_sha` section of the `cmsis_importer.json` file.

That is a bit confusing. @AnotherButler what do you think?

Comment (AnotherButler) [reply docs/reference/contributing/target/rtos.md]: @bulislaw What part do you think is confusing?

File: docs/reference/contributing/target/rtos.md (hunk 3 above)
Comment (deepikabhavnani) [reply docs/reference/contributing/target/rtos.md]: @bulislaw - That change was review comment by @sg- https://github.com/ARMmbed/Handbook/pull/340/files/d827afd63fab6edc4021eec3ddce567217a229d6#diff-d3a387b212d57c83382162e66ce314a0R3

File: docs/reference/contributing/target/rtos.md (hunk 6 above)
Comment (bulislaw) [reply docs/reference/contributing/target/rtos.md]: Maybe confusing isn't the best word for that, but we repeat ourselves and it makes it feel like we talk about something else (?)  `we maintain a small set of changes. We maintain changes` if I'm saying gibberish please ignore :)

File: docs/reference/contributing/target/rtos.md (hunk 3 above)
Comment (bulislaw) [reply docs/reference/contributing/target/rtos.md]: Fair enough.

**Pull request merged**